test/lib.bash

```diff
@@ -45,24 +45,27 @@ function run_e2e_tests {
     && return 1
 }
 
-function run_knative_serving_tests {
-  (
+function checkout_knative_serving {
   local knative_version=$1
   # Setup a temporary GOPATH to safely check out the repository without breaking other things.
   local tmp_gopath
   tmp_gopath="$(mktemp -d -t gopath-XXXXXXXXXX)"
-  cp -r "$GOPATH/bin" "$tmp_gopath"
+  if [[ -d "$GOPATH/bin" ]]; then
+    cp -r "$GOPATH/bin" "$tmp_gopath"
+  fi
   export GOPATH="$tmp_gopath"
 
-  # Save the rootdir before changing dir
-  rootdir="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
-
   # Checkout the relevant code to run
   mkdir -p "$GOPATH/src/knative.dev"
-  cd "$GOPATH/src/knative.dev" || return $?
-  git clone -b "release-${knative_version}" --single-branch https://github.com/openshift/knative-serving.git serving
-  cd serving || return $?
+  pushd "$GOPATH/src/knative.dev" || return $?
+  git clone -b "release-${knative_version}" --depth 1 https://github.com/openshift/knative-serving.git serving
```

Comment:
You can clone directly. No need to have 2 pushed and popd. One is sufficient.

Reply:
Fixed.